### PR TITLE
fix: #3015 Functional pagination

### LINKED ISSUE #3015: Study Page - Implement pagination for table

### PR DESCRIPTION
**Pull request by francisl**
closes #3015 

Functional pagination

### CODE REVIEW

File: src/pages/studies/StudyTableContainer.tsx [hunk 1]
```diff
@@ -23,11 +23,20 @@ const mapState = (state: RootState): StudyTableContainerState => ({
 const connector = connect(mapState, mapDispatch);
```
Comment (evans-g-crsj) [inline src/pages/studies/StudyTableContainer.tsx:23]: I would change the name of the file (remove "Container")

File: src/pages/studies/StudyTableContainer.tsx [hunk 2]
```diff
@@ -23,11 +23,20 @@ const mapState = (state: RootState): StudyTableContainerState => ({
...
+};
 
-type Props = StudiesResults & PropsFromRedux;
+type Props = StudiesResults & PropsFromRedux & PaginationType;
 
 const StudyTable: FC<Props> = (props) => {
-  if (props.loading) {
+  const { loading, pagination } = props;
+  if (loading) {
     return null;
```
Comment (evans-g-crsj) [inline src/pages/studies/StudyTableContainer.tsx:40]: don't we want to display a <Spin/> or else?

File: src/pages/studies/studies.tsx [hunk 3]
```diff
@@ -13,9 +13,12 @@ import styles from './studies.module.scss';
...
+const Studies = () => {
   const { filters } = useFilters();
-  let studiesResults = useGetStudiesPageData({ sqon: filters, first: 10, offset: 0 });
+  const [currentPage, setCurrentPage ] = useState(0);
+
+  let studiesResults = useGetStudiesPageData({ sqon: filters, first: 2, offset: currentPage });
+
   let studiesMappingResults = useGetExtendedMappings('studies');
 
   if (studiesResults.loading || studiesMappingResults.loadingMapping) {
```
Comment (evans-g-crsj) [inline src/pages/studies/studies.tsx:24]: I'm wondering if this logic should be taken care in a non ui part ?

File: src/pages/studies/utils.ts [hunk 4]
```diff
@@ -110,13 +110,20 @@ export const updateQueryParam = (history: any, key: string, value: any): void =>
...
+
     return;
   }
   if (isEmpty(value) && query[key]) {
+    console.log('query : delete');
+
     delete query[key];
   } else {
     query[key] = typeof value === 'object' ? JSON.stringify(value) : value;
+    console.log('query : ', query);
```
Comment (evans-g-crsj) [inline src/pages/studies/utils.ts:123]: remove  console

**Review by evans-g-crsj — CHANGES_REQUESTED** [review]

File: src/pages/studies/StudyTableContainer.tsx (hunk 1 above)
Comment (francisl) [reply src/pages/studies/StudyTableContainer.tsx:23]: this comment would have be more useful 1 month ago when it was merge, but anyway, I'll chage it

File: src/pages/studies/StudyTableContainer.tsx (hunk 2 above)
Comment (francisl) [reply src/pages/studies/StudyTableContainer.tsx:40]: Unrelated to the change.
But now one state when empty and it keeps the previous data while loading as spected.

File: src/pages/studies/studies.tsx (hunk 3 above)
Comment (francisl) [reply src/pages/studies/studies.tsx:24]: this is UI concerned as well

File: src/pages/studies/utils.ts (hunk 4 above)
Comment (francisl) [reply src/pages/studies/utils.ts:123]: fixed

**Review by evans-g-crsj — APPROVED** [review]

**Pull request merged**